scrape/manager_test.go

```diff
@@ -703,3 +710,79 @@ scrape_configs:
 	reload(scrapeManager, cfg2)
 	require.ElementsMatch(t, []string{"job1", "job3"}, scrapeManager.ScrapePools())
 }
+
+func TestManagerScrapeCreatedTimestamp(t *testing.T) {
+	app := &collectResultAppender{}
+	scrapeManager := NewManager(&Options{
+		EnableProtobufNegotiation: true,
+	}, log.NewLogfmtLogger(os.Stderr), &collectResultAppendable{app})
+
+	require.NoError(t, scrapeManager.ApplyConfig(&config.Config{
+		GlobalConfig: config.GlobalConfig{
+			// Extremely high scrape interval, to ensure the only chance to see the
+			// sample is on start and stopAfterScrapeAttempt.
+			ScrapeInterval: model.Duration(5 * time.Second),
+			ScrapeTimeout:  model.Duration(5 * time.Second),
+		},
+		ScrapeConfigs: []*config.ScrapeConfig{{JobName: "test"}},
+	}))
+
+	metric := &dto.MetricFamily{
+		Name: proto.String("expected_metric"),
+		Metric: []*dto.Metric{
+			{
+				Counter: &dto.Counter{
+					Value:            proto.Float64(1.0),
+					CreatedTimestamp: &timestamppb.Timestamp{Seconds: 1, Nanos: 1},
+				},
+			},
+		},
+	}
+
+	b, err := proto.Marshal(metric)
+	require.NoError(t, err)
+
+	// Start fake HTTP target to scrape returning a single metric.
+	server := httptest.NewServer(
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", `application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited`)
+			w.Write(b)
+		}),
+	)
+	defer server.Close()
+
+	serverURL, err := url.Parse(server.URL)
+	require.NoError(t, err)
+
+	// Add fake target directly into tsets + reload. Normally users would use
+	// Manager.Run and wait for minimum 5s refresh interval.
+	scrapeManager.updateTsets(map[string][]*targetgroup.Group{
+		"test": {
+			{
+				Targets: []model.LabelSet{{
+					model.SchemeLabel:  model.LabelValue(serverURL.Scheme),
+					model.AddressLabel: model.LabelValue(serverURL.Host),
+				}},
+			},
+		},
+	})
+	scrapeManager.reload()
+	// Wait for one scrape
+	time.Sleep(5 * time.Second)
+
+	// Currently failing with level=debug scrape_pool=test target=http://127.0.0.1:61076 msg="Append failed" err="proto: wrong wireType = 7 for field Name"
+	scrapeManager.Stop()
+	require.Equal(t, 2, countFloatSamples(app, "expected_metric"))
+}
+
+func countFloatSamples(a *collectResultAppender, expectedMetricName string) (count int) {
+	a.mtx.Lock()
```

Comment:
I think the point of this locking is to avoid fetching while data is being added -- I think you want to add the mutex  on commit like Bartek did in his original PR (since `a.result*` is mutated): https://github.com/prometheus/prometheus/pull/12687/files#diff-368a515513ea20b588ef8c19062e7c3c7a9cc6bbaa890ba7cef49bf5a4642706